### PR TITLE
Convert _project_banner.erb to Phlex

### PR DESCRIPTION
**Pull request by mo-nathan**
## Summary

Converts `app/views/controllers/application/content/_project_banner.erb` to a Phlex component.

### Changes

**Component Implementation** (`app/components/project_banner.rb`):
- Renders project banner with title, location, date range, and edit icons
- Accepts `project` prop directly instead of relying on `content_for` blocks
- Conditionally renders location and date_range based on project data
- Uses `on_project_page` prop to determine title element ID

**Helper Updates** (`app/helpers/tabs/projects_helper.rb`):
- `add_project_banner` now returns the project object
- Views set `@banner_project` from return value
- Eliminates Rails/HelperInstanceVariable RuboCop violation

**Test Coverage** (`test/components/project_banner_test.rb`):
- 8 component tests covering all rendering scenarios
- Tests for projects with/without locations
- Tests for projects with/without date ranges
- Tests for conditional title ID based on controller context

**View Updates**: Updated 13 views across multiple controllers:
- `checklists/show`
- `field_slips/index`
- `observations/index`
- `projects/show`, `projects/locations/index`, `projects/members/index`
- `projects/aliases/*` (index, show, edit, new)
- `projects/violations/index`
- `species_lists/index`, `species_lists/show`

**Fixture Updates**:
- Added banner images to `albion_project` and `pinned_date_range_project` for testing

### Manual Testing Suggestions

1. **Project with location** - Visit a project with a location (e.g., Albion Project)
   - Verify location link appears in banner
   - Click location link and confirm it navigates correctly

2. **Project with date range** - Visit a project with start/end dates
   - Verify date range displays in banner
   - Check formatting is correct

3. **Project show page** - Visit `/projects/{id}`
   - Verify title element has `id="title"`
   - Verify edit icons appear for project owners

4. **Non-project pages with project banner** - Visit observations/species lists filtered by project
   - Verify title element has `id="banner_title"` (not "title")
   - Verify project banner appears at top of page

5. **Project without location or dates** - Visit a basic project
   - Verify banner still renders with title
   - Verify no location/date elements appear

6. **Project tabs** - On any project-related page
   - Verify tabs render correctly below banner
   - Test tab navigation

7. **Responsive behavior** - Test on mobile/tablet
   - Verify banner image scales appropriately
   - Check text overlay remains readable

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: test/components/project_banner_test.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  def render_banner(is_project: false)
+    render(Components::ProjectBanner.new(is_project: is_project))
+  end
+
+  # Helper to render banner with content_for blocks set
+  # (simulates how helpers like add_project_banner set content_for)
+  def render_banner_with_content(project:, with_location: false,
+                                 with_date_range: false,
+                                 is_project: false)
+    wrapper = Class.new(Phlex::HTML) do
```
Comment (mo-nathan) [inline test/components/project_banner_test.rb]: @nimmolo Does this seem reasonable to you or is CC cheating?

**Review by JoeCohen — APPROVED** [review]
Looks good to me. 
FYI: I noticed one preexisting bizarre edge-case bug which IMO is outside the scope of this PR.  [Dragging/Selecting folder instead of an Image #3713](https://github.com/MushroomObserver/mushroom-observer/issues/3713)

File: app/components/project_banner.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        b do
+          a(href: location_path(@project.location.id)) do
+            @project.place_name
+          end
+        end
+      end
+    end
+
+    def render_project_date_range
+      return unless @project&.start_date && @project.end_date
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/components/project_banner.rb]: Meh

**Review by nimmolo — CHANGES_REQUESTED** [review]

File: app/views/controllers/checklists/show.html.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <%
 container_class(:full)
-add_project_banner(@project) if @project
+@banner_project = add_project_banner(@project) if @project
```
[the remark this answers is not in the document]
Comment (nimmolo) [reply app/views/controllers/checklists/show.html.erb]: I see what's up. There are two versions of a project header that could be printed in the page, depending on whether the project ALSO has a banner. That's why the `content_for(:banner_title)`, `content_for(:project_location)`, `content_for(:project_date_range)`, `content_for(:banner_image)` are defined separately.

I feel like a cleaner refactor would be to move all of this logic into the component, and output either a regular project header with these attributes, or a project banner header. 

Or simplest yet - if `@project` is defined, just output a banner in either case. If the project doesn't have a banner image, give that banner div a `background-color: $primary`, but exactly the same layout, with the title bottom-left. Much less complicated.

**Review by nimmolo — DISMISSED** [review]
See new comment in `projects_helper.rb`

Can propose fix in a side PR branch.

Basic description of the issue: Consider a SpeciesList #show view. If `@project` is set (meaning the List is being viewed in the context of a Project), the project association will be reflected in the page header _**one of two ways**_:
- `@project.banner_image?` Project banner is printed, including project `location` and `date_range`. Page title, below banner, does not repeat this info. 
- `!@project.banner_image?` No project banner is printed. On `main`, the page title is printed along with the project `location` and `date_range`. On this branch, only the title is printed.

File: app/views/controllers/projects/aliases/edit.html.erb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <%
-add_project_banner(@project_alias.project)
+@project = @project_alias.project
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/views/controllers/projects/aliases/edit.html.erb]: Done

File: app/helpers/tabs/projects_helper.rb [hunk 5]
```diff
@@ -80,20 +80,10 @@ def add_project_banner(project)
...
-        end
-      end
-
-      if project.start_date && project.end_date
-        content_for(:project_date_range) do
-          tag.b(project.date_range)
-        end
-      end
-
       add_banner_image(project.image)
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/helpers/tabs/projects_helper.rb]: No longer relevant.

**Comment by nimmolo** [comment]
@mo-nathan This looks good. 
Still need to remove conditional project content in `_page_title.erb` lines 5-6 and 23-30. 
```ruby
project_subtitle = content_for?(:project_location) ||
                   content_for?(:project_date_range)
```
```erb
      <% if !has_banner && project_subtitle %>
        <%= tag.div(class: "pl-3 mb-3") do
          [
            tag.div(yield(:project_location), class: "project_location"),
            tag.div(yield(:project_date_range), class: "project_date_range")
          ].safe_join
        end %>
      <% end %>
```
(It won't render anyway, but would be good to clean up.)

File: app/components/project_banner.rb [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+          "species_lists"
+        )
+      ]
+    end
+
+    def tab_item(text, path, controller_name)
+      li(class: "nav-item") do
+        a(href: path, class: tab_classes(controller_name)) { text }
+      end
+    end
```
Comment (nimmolo) [inline app/components/project_banner.rb]: Isn't this duplicating `InternalLink.tab`?

**Review by nimmolo — APPROVED** [review]
